### DISCUSSION TITLE
error while loading libssl

### DISCUSSION THREAD

**Discussion by Tropaion** [dani-garcia/vaultwarden#3833] [Q&A] [answered]
### Subject of the issue

Hello,
I'm just updated to the latest vaultwarden version but when starting it, I get this error, which before never occured:
```error while loading shared libraries: libssl.so.3: cannot open shared object file: No such file or directory```
I have installed openssl and libssl-dev and it always worked, just not with this version.

### Deployment environment
Debian 11 LXC on Proxmox

* vaultwarden version: 1.29.2

* Install method: built from source: https://wiki.flowerhouse.at/index.php/BitWarden#Building

* Clients used: 
Web Vaul + Android + Desktop

* Reverse proxy and version: 
Caddy

* MySQL/MariaDB or PostgreSQL version: 
MySQL

* Other relevant details:
Building and deployment on separate LXCs

**Comment by BlackDex** [comment, marked as answer]
If you build it your self, i suggest to use the `vendored_openssl` feature. That should probably fix your issues.

Though, it should just work out-off-box and should detect the OpenSSL version and dev packages on the host you build it on.

If you build the binary on a different system then you run it on, that will cause issues, you need to make sure the libraries match.

**Reply by Tropaion** [reply]
Thank you, my mistake, I found the reason. I accidently used the new debian version to build.
